--- src/tester.c.orig	2016-12-13 23:57:45.038832795 +0100
+++ src/tester.c	2016-12-14 00:50:20.386929114 +0100
@@ -59,6 +59,8 @@
 #ifdef __linux
 /*for monitoring total space allocated via malloc*/
 #include <malloc.h>
+#include <sys/time.h>
+#include <sys/resource.h>
 #endif
 
 #ifndef F_OK
@@ -294,12 +296,15 @@
 	 * mallinfo() itself is the less worse solution. Allocated bytes are returned as 'int' so limited to 2GB
 	 */
 	if (max_vm_kb) {
-		struct mallinfo minfo = mallinfo();
-		if (minfo.uordblks > max_vm_kb * 1024) {
+		struct rusage usage;
+
+		getrusage(RUSAGE_SELF, &usage);
+
+		if (usage.ru_maxrss > max_vm_kb * 1024) {
 			bc_tester_printf(
 				bc_printf_verbosity_error,
 				"The program exceeded the maximum amount of memory allocatable (%i bytes), aborting now.\n",
-				minfo.uordblks);
+				usage.ru_maxrss);
 			abort();
 		}
 	}
@@ -382,10 +387,6 @@
 			}
 		}
 	}
-#ifdef __linux
-	bc_tester_printf(bc_printf_verbosity_info, "Still %i kilobytes allocated when all tests are finished.",
-					 mallinfo().uordblks / 1024);
-#endif
 
 	return CU_get_number_of_tests_failed()!=0;
 
